updater/lib/dependabot/updater/operations.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require "dependabot/updater/operations/update_all_versions"
+
+# This module is responsible for determining which Operation a Job is requesting
+# the Updater to perform.
+#
+# The design goal for this module is to make these classes easy to understand,
+# maintain and extend so we can eventually support community-contributed
+# alternatives or ecosystem-specific implementations.
+#
+# Consider the following guidelines when working on Operation classes:
+#
+# - Operations act on the principal of least knowledge prefering to cherry pick
+#   arguments from the Dependabot::Job so their contract is explicit.
+#
```

Comment:
I staked out these rules before I started writing any code and I wasn't able to keep to this one.

I'm very concerned at the `Dependabot::Job` becoming too broad an AI, it already provides vulnerability checking, etc, so I think there is work to be done in narrowing it into a few smaller interfaces.

I think solving this is fairly large and off the critical path, so I'm leaning towards backlogging this until we get a bit further with other priorities.